### PR TITLE
chore: bump core v1.31.0-tm-v0.34.29

### PR DESCRIPTION
**Pull request by evan-forbes**
## Overview

updates to the latest version of core

modifies `make mod` to automate calling `go work sync`, and then `go mod tidy` twice, once for the root, and then again for the testground subdir

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -41,6 +41,8 @@ install: go.sum
 mod:
 	@echo "--> Updating go.mod"
 	@go mod tidy
+	@echo "--> Updating go.mod in ./test/testground"
+	@(cd ./test/testground && go mod tidy)
```
Comment (staheri14) [inline Makefile:45]: 👍

**Review by staheri14 — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (rootulp) [reply Makefile:45]: do we want a `go work sync` prior to both of these commands?

**Review by rootulp — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (cmwaters) [reply Makefile:45]: Yeah makes sense

**Review by cmwaters — DISMISSED** [review]
I see you weren't going to wait for my state sync fix 👀

**Comment by evan-forbes** [comment]
> I see you weren't going to wait for my state sync fix 👀

super sorry about that, I didn't see it until this morning!

File: Makefile (hunk 1 above)
Comment (evan-forbes) [reply Makefile:46]: updated in https://github.com/celestiaorg/celestia-app/pull/2979/commits/ac5bde2528b3c25110288fe98ed38262970a88b1

**Review by rootulp — APPROVED** [review]

**Review by staheri14 — APPROVED** [review]

**Pull request merged**